integration/test/performance_tracker/transmitter/transmitter.go

```
@@ -0,0 +1,237 @@
+		AttributeDefinitions: []*dynamodb.AttributeDefinition{
+			{
+				AttributeName: aws.String("Hash"),
+				AttributeType: aws.String("S"),
+			},
+		},
+		KeySchema: []*dynamodb.KeySchemaElement{
+			{
+				AttributeName: aws.String("Hash"),
+				KeyType:       aws.String("HASH"),
+			},
+		},
+		ProvisionedThroughput: &dynamodb.ProvisionedThroughput{
+			ReadCapacityUnits:  aws.Int64(10),
+			WriteCapacityUnits: aws.Int64(10),
+		},
+		TableName: aws.String(transmitter.DataBaseName),
+	} // this is the config for the new table
+	
+	_, err := transmitter.DynamoDbClient.CreateTable(input)
+	if err != nil {
+		fmt.Printf("Got error calling CreateTable: %s", err)
+		return err
+	}
+	
+	fmt.Println("Created the table", transmitter.DataBaseName)
+	return nil
+}
+/*
+AddItem() 
+Desc: Takes in a packet and
+will convert to dynamodb format  and upload to dynamodb table.
+Param: 
+	packet * map[string]interface{}:  is a map with data collection data
+Side effects: 
+	Adds an item to dynamodb table
+*/
+func (transmitter * TransmitterAPI) AddItem(packet map[string]interface{})(string,error){
+	// fmt.Printf("Packet: %+v \n",packet)
+	// metrics, _ := dynamodbattribute.MarshalMap(packet.Metrics)
+	// fmt.Printf("Metric: %+v \n OG:%+v \n",metrics, packet.Metrics)
+	item, err := dynamodbattribute.MarshalMap(packet)
+	if err != nil {
+		panic(err)
+	}
+	DBitem := &dynamodb.PutItemInput{
+		Item:      item,
+		TableName: aws.String(transmitter.DataBaseName),
+	}
+	// fmt.Println(DBitem)
+	// return err
+	_, err = transmitter.DynamoDbClient.PutItem(DBitem)
+	if err != nil {
+		fmt.Printf("Couldn't add item to table. Here's why: %v\n", err)
+	}
+	return fmt.Sprintf("%v",item),err
+}
+/*
+TableExist()
+Desc: Checks if the the table exist and returns the value
+//https://github.com/awsdocs/aws-doc-sdk-examples/blob/05a89da8c2f2e40781429a7c34cf2f2b9ae35f89/gov2/dynamodb/actions/table_basics.go
+*/
+func (transmitter * TransmitterAPI) TableExist() (bool,error){
```

Do we need to return an error here? Looks like we just need to return a bool

---

Sounds other way around to me. Do we need to return a bool here? An error is fine by me 👍

---

No, the name of the function implies that the purpose of the function is to check whether or not the table exists. That means it makes sense to be a bool. If the table doesn't exist, want to try to create it, so it doesn't make much of a difference if we don't find the table or if there's an error trying to look it up. It could help short-circuit the logic so if there is an error we fail fast, but I'm honestly not really concerned about that at all.